hera_pspec/uvpspec.py

```diff
@@ -2088,9 +2089,23 @@ def combine_uvpspec(uvps, merge_history=True, verbose=True):
     if store_window:
         u.window_function_array = odict()
     if store_cov:
-        u.cov_array = odict()
-        # cov_model will track whether error bars are from cmobination of techniques
-        u.cov_model = ','.join([uvp.cov_model for uvp in uvps])
+        # ensure cov model is the same for all uvps
+        if len(set([uvp.cov_model for uvp in uvps])) > 1:
+            store_cov = False
+        else:
+            u.cov_array = odict()
+            u.cov_model = uvps[0].cov_model
+    if store_stats:
+        # get shared stats keys
+        stored_stats = [set(uvp.stats_array.keys()) for uvp in uvps]
+        for i in range(len(stored_stats)):
+            stored_stats[0].intersection_update(stored_stats[i])
+        stored_stats = stored_stats[0]
+        # if nothing in common, set to False
+        if len(stored_stats) == 0:
+            store_stats = False
+        u.stats_array = odict([(stat, odict()) for stat in stored_stats])
```

Comment:
Maybe better to add an `else' block here to avoid constructing a redundant u.stats_array when len(stored_stats) == 0

Reply:
ah, good catch